### PR TITLE
Feature/plugin toggle

### PR DESCRIPTION
**Pull request by pieroit**
# Description

 - plugin switch on/off
 - plugin settings
 - tests for both

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/tests/infrastructure/test_package.py [hunk 1]
```diff
@@ -8,7 +8,9 @@ def test_unpackage():
     
     zip_path = create_mock_plugin_zip()
     zip = Package(zip_path)
-    zip.unpackage("tests/infrastructure/")
+    extracted = zip.unpackage("tests/infrastructure/")
```
Comment (Pingdred) [inline core/tests/infrastructure/test_package.py:11]: Shouldn't it be extracted to `tests/mocks/mock_plugin_folder/`?

File: core/cat/infrastructure/package.py [hunk 2]
```diff
@@ -19,13 +20,26 @@ def __init__(self, path):
...
         elif self.extension == 'tar':
             with tarfile.open(self.path, 'r') as tar_ref:
                 tar_ref.extractall(to)
 
+        # list of folder contents after extracting
+        ls_after = os.listdir(to)
+
+        # return extracted contents paths
+        # TODO: does not handle overwrites, should extract in a /temp new folder and then copy to destination
+        ls = set(ls_after) - set(ls_before)
```
Comment (zAlweNy26) [inline core/cat/infrastructure/package.py:40]: Everything seems fine to me, I just don't understand this part

**Review by zAlweNy26 — APPROVED** [review]
Everything seems fine to me.

**Review by Pingdred — APPROVED** [review]

File: core/cat/infrastructure/package.py (hunk 2 above)
Comment (pieroit) [reply core/cat/infrastructure/package.py:40]: `ls_before` = what was in the destination folder before extraction
`ls_after` = what was after
`ls` = list of contents extracted

Can be done better, it works but does not take into account if some contents are overwritten

Comment (pieroit) [reply core/cat/infrastructure/package.py:40]: `set` in python is a native data structure representing a set of unique elements
`setA - setB` returns elements in setA that are not present in setB

File: core/tests/infrastructure/test_package.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pieroit) [reply core/tests/infrastructure/test_package.py:11]: yes

File: core/cat/mad_hatter/core_plugin/hooks/plugin_settings.py [hunk 3]
```diff
@@ -4,47 +4,77 @@
...
+    #    length: int  # required field, type int
+    #    description: str = "my fav cat" # optional field, type str, with a default
 
 
 @hook(priority=0)
-def plugin_settings_save(settings):
-    # this hook passes the plugin settings as sent to the http endpoint (via admin, or any client)
-    # the settings to save are validated according to the json schema given above
-    # default behavior for this hook, just saves contents in a settings.json in the plugin folder
+def plugin_settings_get_schema():
```
[the remark this answers is not in the document]
Comment (pieroit) [reply core/cat/mad_hatter/core_plugin/hooks/plugin_settings.py:16]: ok

File: core/tests/infrastructure/test_package.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pieroit) [reply core/tests/infrastructure/test_package.py:11]: moving also `metadata.test.json` into mocks folder

**Pull request merged**